concourse/steps/version.mako

```diff
@@ -18,6 +18,7 @@ legacy_version_file = os.path.join(job_step.output('version_path'), 'number')
 if (read_callback := version_trait.read_callback()):
   read_callback = os.path.join(main_repo.resource_name(), read_callback)
 
+write_callback = None
```

Comment:
This is not the issue, as `write_callback` is defined by using the "walrus" operator.

The issue lies in l. 107 and the usage of `write_callback` therein. I'll update the PR.

Reply:
https://peps.python.org/pep-0572/ .. :-)

Reply:
Hey thanks guys! I'm always up for learning. Thanks a bunch for the leads! 😄